api/features_api.py

```diff
@@ -29,9 +29,25 @@ class FeaturesAPI(basehandlers.APIHandler):
 
   def do_get(self):
     user = users.get_current_user()
-    feature_list = models.Feature.get_chronological(
-        version=2,
-        show_unlisted=permissions.can_edit_feature(user, None))
+    feature_list = None
+
+    # Query-string parameter 'milestone' is provided
+    if self.request.args.get('milestone') is not None:
+      try:
+        milestone = int(self.request.args.get('milestone'))
+        feature_list = models.Feature.get_in_milestone(
+          version=2,
+          show_unlisted=permissions.can_edit_feature(user, None), 
+          milestone=milestone)
+      except ValueError:
+        self.abort(400, msg='Invalid  Milestone')
+
```

Comment:
If the query string is anything other than a number, the request will be aborted with error code 400.

Reply:
That sounds right.